### DISCUSSION TITLE
Inexplicable Selectors from Preflight

### DISCUSSION THREAD

**Discussion by zsteiner** [tailwindlabs/tailwindcss#9166] [Help]
I'm looking at the compiled CSS from Tailwind v3.1.4 and every selector is appended with `:not(#\#)`. I'm familiar with `:not()`, but not at all sure what `#/#` is and why it's getting added to every selector. These are special characters that break Google and Github search, so searching for what this means or where it comes from has been challenging. I have looking Tailwind and Postcss source code and don't see anything that adds this. Because of this, these selectors have more specificity than they should as resets. I can not for the life of me find out where this added pseudo-class is coming from. Any ideas?

```
:not(#\#),
:not(#\#):after,
:not(#\#):before
{
  box-sizing:border-box;
  border: 0 solid #e1e9f0
}
```

**Comment by Palmieri78759** [comment]
I am also confused and in need of light on this same issue. Need help. [PeopleNet PFM](https://www.pfmlogin.net/)

**Comment by wongjn** [comment]
Got a repository that reproduces the issue that we can look at?

**Reply by zsteiner** [reply]
I sure do. In streamlining the private repo down for a public repro, I think I found the culprit. https://github.com/clearbit/test-css/blob/5f580a9c586a504dd1269bbfaee8b17937122339/postcss.config.js#L8-L10

I was trying to get nesting to work correctly and add this config to my `postcss.config.js`:

```javascript
'postcss-preset-env': {
  features: { 'nesting-rules': false },
},
```

Building CSS with that config has the appended the selector, but removing the config removes the selector!

**Reply by zsteiner** [reply]
Follow up. https://tailwindcss.com/docs/using-with-preprocessors#nesting Following these docs broke `& { }` nesting. I had to add `postcss-nested: {}` to restore nested selectors using &.
